app/src/main/java/com/example/dogbook/main/fragments/TimelineFragment.java

```diff
@@ -78,23 +81,44 @@ private void setUpRecyclerView() {
 
 
     public void refreshPosts() {
-        ParseQuery<Post> query = ParseApplication.getAllPostsQuery();
-        query.findInBackground(new FindCallback<Post>() {
+
+        HashMap<String, Object> params = new HashMap<String, Object>();
+        ParseCloud.callFunctionInBackground("getTimeline", params, new FunctionCallback<Object>() {
             @Override
-            public void done(List<Post> objects, ParseException e) {
+            public void done(Object object, ParseException e) {
                 if (e == null) {
+                    HashMap results = (HashMap) object;
                     posts.clear();
-                    posts.addAll(objects);
+                    posts.addAll((Collection<? extends Post>) results.get("posts"));
+                    posts = Post.addReactions(posts, (List<HashMap>) results.get("reactions"));
                     postsAdapter.notifyDataSetChanged();
                     rvPosts.smoothScrollToPosition(0);
                     ptrContainer.setRefreshing(false);
 
                     return;
                 }
-                Log.e(TAG, "Issue finding posts in Parse", e);
-                Toast.makeText(getContext(), "Unable to refresh posts", Toast.LENGTH_SHORT).show();
-                ptrContainer.setRefreshing(false);
+                Log.e(TAG, "Timeline not fetched", e);
+
             }
         });
+
+//        ParseQuery<Post> query = ParseApplication.getAllPostsQuery();
+//        query.findInBackground(new FindCallback<Post>() {
+//            @Override
+//            public void done(List<Post> objects, ParseException e) {
+//                if (e == null) {
+//                    posts.clear();
+//                    posts.addAll(objects);
+//                    postsAdapter.notifyDataSetChanged();
```

Comment:
Forgot to delete comments, will do in next PR

Reply:
Let's not merge next time if there's code that's commented out